test/Compressors/sampling.jl

```diff
@@ -0,0 +1,501 @@
+module Sampling_compressor
+using Test, RLinearAlgebra, Random
+using StatsBase: ProbabilityWeights, sample
+import LinearAlgebra: mul!, Adjoint
+using ..FieldTest
+using ..ApproxTol
+
+Random.seed!(2131)
+@testset "Sampling" begin
+    @testset "Sampling: Compressor" begin
+        # Verify Supertype
+        @test supertype(Sampling) == Compressor
+
+        # Verify fields and types
+        @test fieldnames(Sampling) == (:cardinality, :compression_dim, :distribution)
+        @test fieldtypes(Sampling) == (Cardinality, Int64, Distribution)
+
+        # Verify the Internal Constructor
+        let cardinality = Left(), compression_dim = 0, distribution = Uniform()
+            @test_throws ArgumentError(
+                "Field `compression_dim` must be positive."
+            ) Sampling(
+                cardinality, compression_dim, distribution
+            )
+        end
+
+        let cardinality = Undef(), compression_dim = 2, distribution = Uniform()
+            @test_throws ArgumentError(
+                "`cardinality` must be specified as `Left()` or `Right()`.\
+                `Undef()` is not allowed in `CountSketch` structure."
+            ) Sampling(
+                cardinality, compression_dim, distribution
+            )
+        end
+
+        # Default values
+        let sc_default = Sampling()
+            @test sc_default.cardinality == Left()
+            @test sc_default.compression_dim == 2
+            @test sc_default.distribution isa Uniform
+            # Check defaults of the default Uniform distribution instance
+            @test sc_default.distribution.cardinality == Undef()
+            @test sc_default.distribution.replace == false
+        end
```

Comment:
Something here does not look right to me. By default, the distribution is `Uniform` and being implemented with a cardinality of `Undef()`. That means, `complete_distribution` and `update_distribution` which are used in `complete_compressor` and `update_compressor` will throw an error for the default. The default should not be `Undef()`, it should match whatever we choose as the default in `Sampling()`.

@dmaldona what do you think? 

@nathanielpritchard this to me seems like a high level design concern given that the default argument does not work. My other concern is if there are other distributions, do they all have to have a cardinality argument? If so, then there is an easy fix. If not, then this is a problem.